### PR TITLE
Feature/#34 Add changelog

### LINKED ISSUE #34: Add changelog

### LINKED ISSUE #45: Automate PyPi version bumping

### PR DESCRIPTION
**Pull request by 49e94b8f256530dc0d41f740dfe8a4c1**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -15,22 +15,53 @@ pipeline {
...
+                )}"""
+            }
+            when {
+                branch "develop"
+            }
+            steps {
+                sh 'echo ${VERSION}'
+                sh 'git config user.name jenkins'
+                sh 'git config user.email kenokech94@gmail.com'
+                sh 'git stash save setup.py'
```
Comment (zawi99) [inline Jenkinsfile:51]: Could you tell me why you stash changes here?

Comment (49e94b8f256530dc0d41f740dfe8a4c1) [reply Jenkinsfile:51]: The real question is why is `develop` checked out twice? Nice catch.

https://github.com/project-mk-ultra/torrent-tracker-scraper/blob/dfa0794d7c3b47844acf6281f32db0a493a41d9a/Jenkinsfile#L52

Comment (49e94b8f256530dc0d41f740dfe8a4c1) [reply Jenkinsfile:51]: I remember now. This code is not running on Jenkins. It's running inside the Docker container. Beforehand Jenkins does this 

```
Checking out Revision 146cda170956dd812ed9ee400e7e602917ce1d40 (refs/remotes/origin/feature/#34-add-changelog)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 146cda170956dd812ed9ee400e7e602917ce1d40 # timeout=10
Commit message: "new: dev: add changelog"
 > git rev-list --no-walk e08e3681bf2aa5f5213c67060304893c373d2905 # timeout=1
```

Perhaps there is a better way of doing this.

**Comment by 49e94b8f256530dc0d41f740dfe8a4c1** [comment]
Perhaps it would be wise to amend your last commit, making the message friendly to gitchangelog

    chg: dev: jenkins file formatting !cosmetic 

Reference: https://github.com/vaab/gitchangelog/blob/master/src/gitchangelog/gitchangelog.rc.reference

**Comment by zawi99** [comment]
But do we really want commits like that (not really relevant) to be included into changelog?

**Comment by 49e94b8f256530dc0d41f740dfe8a4c1** [comment]
@zawi99 You can squash the commits before merge if you want.

**Pull request merged**